File: java/src/main/java/deltix/luminary/LiteralBoolean.java
package deltix.luminary;

public class LiteralBoolean extends Literal {
    private boolean value;

    private LiteralBoolean(boolean value) {
        super(LiteralKind.BOOLEAN);
        this.value = value;
    }

    public boolean getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value ? "true" : "false";
    }

    public static final LiteralBoolean TRUE = new LiteralBoolean(true);
    public static final LiteralBoolean FALSE = new LiteralBoolean(false);
}

File: csharp/main/LiteralBoolean.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralBoolean : Literal, ILiteral<Boolean>
	{
		private LiteralBoolean(Boolean value)
			: base(LiteralKind.Boolean)
		{
			Value = value;
		}

		public Boolean Value { get; }

		public override String ToString()
		{
			return Value ? "true" : "false";
		}

		public static readonly LiteralBoolean True = new LiteralBoolean(true);
		public static readonly LiteralBoolean False = new LiteralBoolean(false);
	}
}
